llvm/lib/CAS/LazyMappedFileRegion.cpp

```diff
@@ -242,14 +242,60 @@ Expected<LazyMappedFileRegion> LazyMappedFileRegion::create(
 
   sys::fs::file_t File = sys::fs::convertFDToNativeFile(FD);
 
+  struct FileLockRAII {
+    LazyMappedFileRegion &LMFR;
+    bool IsLocked = false;
+
+    enum LockKind { Shared, Exclusive };
+
+    FileLockRAII(LazyMappedFileRegion &LMFR) : LMFR(LMFR) {}
+    ~FileLockRAII() { unlock(); }
+
+    Error lock(LockKind LK) {
+      unlock();
+      if (std::error_code EC = sys::fs::lockFile(*LMFR.FD, LK == Exclusive))
+        return createFileError(LMFR.Path, EC);
+      IsLocked = true;
+      return Error::success();
+    }
+
+    void unlock() {
+      if (IsLocked) {
+        sys::fs::unlockFile(*LMFR.FD);
```

Comment:
unlock can fail. We should propagate the error, or if we really think it's okay to ignore the return value then we should comment why

Reply:
Good point! See updated commit.